### PR TITLE
Using coro to create a generator

### PR DESCRIPTION
**Pull request by trangdata**
Example use case:

``` r
query_url <- "https://api.openalex.org/works?filter=cites%3AW2755950973"
oar <- oa_generate(query_url)
for (i in seq.int(1000)){
# or something like: while record_i != ".__exhausted__."
  record_i <- oar()
  # processing record_i here
  # for example:
  # saveRDS(record_i, paste0("rec-", i, ".rds"))
}

# or, occasionally, if you know the result is a manageable list that fits in memory
# you can use coro::collect to get the entire list of results
# (although in this case we would recommend oa_request instead)
system.time({
  results <- coro::collect(oar)
})
```

### CODE REVIEW

**Comment by rkrug** [comment]
I personally think this introduces a lot of complexity into the package as well as for the user. It might solve the issue (I don't know because I do not understand Coro), but it is implementation wise heavier from my perspective.

File: R/globals.R
Comment (yjunechoe) [inline R/globals.R:1]: Just curious - what's the purpose of this again? Do we get CRAN notes without putting these variables here? (Just thought I should know for the future)

**Review by yjunechoe — APPROVED** [review]
LGTM!

File: R/globals.R
Comment (trangdata) [reply R/globals.R:1]: That's right! More detailed discussion [here](https://github.com/r-lib/coro/issues/40).

Comment (yjunechoe) [reply R/globals.R:1]: TIL! Thanks!

Comment (trangdata) [reply R/globals.R:1]: It's good that you mentioned it 🙏🏽  — reminded me that we no longer need `oa_email` and `oa_apikey` there after I followed Lionel's suggestion for `oa_generate`.

**Comment by rkrug** [comment]
There is the convenience function `oa_sowball()` which would also profit from something along this line, but I think the implementation could be easier as the input papers need to be split into chunks?

**Pull request merged**